hack/cleane2e/cleane2e.go

```diff
@@ -90,13 +142,73 @@ func run(ctx context.Context, log *logrus.Entry) error {
 		}
 		for _, pls := range plss {
 			for _, peconn := range *pls.PrivateEndpointConnections {
-				log.Printf("deleting private endpoint connection %s/%s/%s", *g.Name, *pls.Name, *peconn.Name)
-				privatelinkservicescli.DeletePrivateEndpointConnection(ctx, *g.Name, *pls.Name, *peconn.Name)
+				log.Infof("deleting private endpoint connection %s/%s/%s", *g.Name, *pls.Name, *peconn.Name)
+				if !dryRun {
+					privatelinkservicescli.DeletePrivateEndpointConnection(ctx, *g.Name, *pls.Name, *peconn.Name)
+				}
 			}
 		}
 
-		log.Printf("deleting resource group %s", *g.Name)
-		resourcegroupscli.Delete(ctx, *g.Name)
+		// list aro clusters if any and append existing app clientID to the delete list
+		aros, err := arocli.ListByResourceGroup(ctx, *g.Name)
+		if err != nil {
+			log.Errorf("%s: %s", *g.Name, err)
+			continue
+		}
+
+		// if there is ARO in the resource group, store the app clientID for later cleanup
+		for _, aro := range aros {
+			clientID := aro.ServicePrincipalProfile.ClientID
+			_, ok := clientIDForDeletion[*clientID]
+			if !ok {
+				clientIDForDeletion[*clientID] = true
+			}
+		}
+
+		log.Infof("deleting resource group %s", *g.Name)
+		if !dryRun {
+			resourcegroupscli.Delete(ctx, *g.Name)
+		}
+	}
+
+	roleAssignments, err := roleassignmentcli.List(ctx, "")
+	if err != nil {
+		log.Errorf("Getting roleAssignments: %s", err)
+		return err
+	}
+
+	// iterate over all roleAssignments and delete the ones associated with deleted ARO clusters
+	for roleAssignments.NotDone() {
+		pageResult := roleAssignments.Values()
+
+		for _, roleAssignmentResult := range pageResult {
+			_, ok := clientIDForDeletion[*roleAssignmentResult.PrincipalID]
```

Comment:
Need to confirm, as this will not work if cluster is gone.

Reply:
We need to brute force this. We list all clusters in sub, and list all roles. And If cluster does not exit, delete Rolebinding. And target only `aro-xxx` rolebindings.